SPEC.md

```diff
@@ -10164,6 +10112,83 @@ Example output:
 </p>
 </details>
 
+### `length`
+
+```
+Int length(Array[X]|Map[X, Y]|Object|Struct|String)
+```
+
+Returns the length of the input argument as an `Int`:
+
+* For an `Array[X]` argument: the number of elements in the array.
+* For a `Map[X, Y]` argument: the number of items in the map.
+* For a `Struct` or `Object` argument: the number of members in the collection.
+* For a `String` argument: the number of characters in the string.
+
+Note that every struct of a given type will always have the same length, equal to the number of members in the struct definition, regardless of whether any of the members are optional.
+
+**Parameters**
+
+1. `Array[X]`|`Map[X, Y]`|`Object`|`Struct`|`String`: A collection or string whose elements are to be counted.
+
+**Returns**: The length of the collection/string as an `Int`.
+
+<details>
+<summary>
+Example: test_length.wdl
+
+```wdl
+version 1.2
+
+struct Name {
```

Comment:
Could you make one of the members of `Name` optional so that the example effectively checks that optional members are also counted?